pkg/storkctl/schedulepolicy.go

```diff
@@ -60,6 +67,219 @@ func newGetSchedulePolicyCommand(cmdFactory Factory, ioStreams genericclioptions
+		},
+	}
+	//Picking up user inputs and setting the defaults for flags
+	createSchedulePolicyCommand.Flags().StringVarP(&schedulePolicyType, "policy-type", "t", "Interval", "Select Type of Schedule Policy to apply. Interval / Daily / Weekly / Monthly. ")
+	createSchedulePolicyCommand.Flags().IntVarP(&intervalMinutes, "interval-minutes", "", 30, "Specify the interval, in minutes, after which Portworx should trigger the operation. ")
+	createSchedulePolicyCommand.Flags().IntVarP(&retain, "retain", "", 0, "For backup operations, specify how many backups Portworx should retain. ")
+	createSchedulePolicyCommand.Flags().StringVarP(&time, "time", "", "12:00AM", "Specify the time of the day in the 12 hour AM/PM format, when Portworx should trigger the operation. ")
+	createSchedulePolicyCommand.Flags().StringVarP(&dailyForceFullSnapshotDay, "force-full-snapshot-day", "", "MONDAY", "For daily scheduled backup operations, specify on which day to trigger a full backup. ")
+	createSchedulePolicyCommand.Flags().StringVarP(&weeklyDay, "weekly-day", "", "Sunday", "Specify the day of the week when Portworx should trigger the operation. You can use both the abbreviated or the full name of the day of the week")
+	createSchedulePolicyCommand.Flags().IntVarP(&monthlyDate, "monthly-date", "", 1, "Specify the day of the month when Portworx should trigger the operation")
+	return createSchedulePolicyCommand
+}
+
+func newDeleteSchedulePolicyCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	deleteSchedulePolicyCommand := &cobra.Command{
+		Use:     schedulePolicySubcommand,
+		Aliases: schedulePolicyAliases,
+		Short:   "Delete schedule policies",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) == 0 {
+				util.CheckErr(fmt.Errorf("at least one argument needs to be provided for schedule policy name"))
+				return
+			}
+
+			namespaces, err := core.Instance().ListNamespaces(nil)
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+			for i := 0; i < len(args); i++ {
+				//ensure that the schedule policy is not being used by any stork resource
+				// (MigrationSchedule / ApplicationBackups / Snapshot Schedule)
+				scheduleTypes := []string{migrationSchedule, applicationBackupSchedule, volumeSnapshotSchedule}
+				policyLinkedTo, err := isSchedulePolicyBeingUsed(scheduleTypes, args[i], namespaces)
+				if err != nil {
+					util.CheckErr(fmt.Errorf("cannot delete the Schedule Policy: %s. Unable to verify if the schedulePolicy is linked to other resources. %w", args[i], err))
+					return
+				}
+				if len(policyLinkedTo[migrationSchedule])+len(policyLinkedTo[applicationBackupSchedule])+len(policyLinkedTo[volumeSnapshotSchedule]) != 0 {
+					// Print the names of resources using the given schedule policy
+					message := "\nThe resource is linked to -> "
+					for _, scheduleType := range scheduleTypes {
+						schedules := policyLinkedTo[scheduleType]
+						if len(schedules) > 0 {
```

Comment:
can add the resource namespace also in the message.

Reply:
Makes sense. Will address in the revision

Reply:
Have updated the name being passed in the helper function